### ISSUE TITLE
Crash on switching folders(or the tabs) or maybe fetch data?

### ISSUE THREAD

**Issue by EdisonJwa** [mock-foundation/moc#179]
I was trying to switch the folders and seems the moc is also fetching the data at this time. and then it crashed

version 0.3.0 (15)

The Problem report is as the following
> Github reports body is too long and I'll put it in the following gist

https://gist.github.com/EdisonJwa/803c22aa179e978e668b29ee681fefb0

**Comment by ggoraa** [comment]
The styling system should support UTF-8 fully, but thanks for clarifying!

**Comment by EdisonJwa** [comment]
And I'm not sure if the lag was also caused by CJK characters

**Comment by ggoraa** [comment]
@EdisonJwa what did you do exactly to trigger that crash? i can succesfully open that channel, and Moc doesn't die

**Comment by ggoraa** [comment]
@EdisonJwa No, lag is definitely not caused by CJK characters, it's a app-wide issue, because of poor optimization

**Comment by EdisonJwa** [comment]
> @EdisonJwa No, lag is definitely not caused by CJK characters, it's a app-wide issue, because of poor optimization

Please check the video
https://user-images.githubusercontent.com/22003734/201720117-97ae52cd-daa6-46b6-8f13-2d54cb090c9e.mov

**Comment by ggoraa** [comment]
Hm
You are on macOS Monterey, I am on Ventura, that's probably the reason why I can't reproduce it
